lib/broadway_rabbitmq/producer.ex

```diff
@@ -17,11 +17,15 @@ defmodule BroadwayRabbitMQ.Producer do
       See `AMQP.Basic.qos/2` for the full list of options. Pay attention that the
       `:global` option is not supported by Broadway since each producer holds only one
       channel per connection.
+    * `:requeue` - Optional. Defines a strategy for requeuing failed messages.
+      Possible values are: `:always` - always requeue, `:never` - never requeue,
+      `:unless_redelivered` - requeue the message unless it's been redelivered.
+      Default is `:never`.
```

Comment:
Is never the best default? I would stick with the AMQP default of `:always`.

Reply:
AMQP doesn't have a default for the requeuing strategy as a whole. It just defines `requeue: true` as default for `Basic.reject/1` so the user is responsible for handling that explicitly with `requeue: false` at some point, for instance, using `redelivered` as in the README.  In our case, we assume this responsibility and provide 3 options. One problem with `:always` is that if the user doesn't configure any TTL, the message will be kept in the queue indefinitely. Maybe `:unless_redelivered` would be a better/safer choice.

@sescobb27 I'd love to hear your thoughts on this one.

BTW, @josevalim how about using `:once` instead of `:unless_redelivered`?